src/replication.c

```diff
@@ -713,15 +800,21 @@ update_replica_entry(spec_t *spec, replica_t *replica, int iofd, char *replicaip
 
 	int rc;
 	pthread_t replica_sender_thread;
-	rc = pthread_mutex_init(&replica->r_mtx, NULL); //check
+	__sync_fetch_and_add(&replica->refcount, 1);
```

Comment:
why do we need this here? we already incremented in create_replica_entry right..?